### PR TITLE
fix(getobjecttree): getobjecttree should work even allocation is empty

### LINKED ISSUE #592: getobjecttree doesn't work if there is no any file is uploaded
```
0box-sdk        [ERROR]  2022/03/16 11:31:40.716329 common.go:35: Object tree response : 400
0box-sdk        [ERROR]  2022/03/16 11:31:40.716480 copyworker.go:102: 400: Object tree error response: Body: {"code":"invalid_parameters","error":"invalid_parameters: Invalid path. Could not find object tree"}
0box-sdk        [ERROR]  2022/03/16 11:31:40.726235 common.go:35: Object tree response : 400
0box-sdk        [ERROR]  2022/03/16 11:31:40.726378 copyworker.go:102: 400: Object tree error response: Body: {"code":"invalid_parameters","error":"invalid_parameters: Invalid path. Could not find object tree"}
```

reproduce it
- ./zbox newallocation
- ./zbox copy --remotpath / --destpath /dir --allocation $alloc

### PR DESCRIPTION
**Pull request by cnlangzi**
### Fixes
- https://github.com/0chain/blobber/issues/592

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/blobber/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- system_test: https://github.com/0chain/system_test/pull/226

### CODE REVIEW

File: code/go/0chain.net/blobbercore/handler/handler.go [hunk 1]
```diff
@@ -153,6 +153,17 @@ func WithStatusConnection(handler common.StatusCodeResponderF) common.StatusCode
 	}
 }
 
+func WithStatusReadOnlyConnection(handler common.StatusCodeResponderF) common.StatusCodeResponderF {
+	return func(ctx context.Context, r *http.Request) (interface{}, int, error) {
+		ctx = GetMetaDataStore().CreateTransaction(ctx)
+		resp, statusCode, err := handler(ctx, r)
+		defer func() {
+			GetMetaDataStore().GetTransaction(ctx).Rollback()
```
Comment (peterlimg) [inline code/go/0chain.net/blobbercore/handler/handler.go:161]: If the transaction was used by other threads and update the database, what would happen by calling the `Rollback()` here? Is it possible the changes will be reset?

**Review by peterlimg — CHANGES_REQUESTED** [review]

**Comment by peterlimg** [comment]
Do we have system test that cover this case

File: code/go/0chain.net/blobbercore/handler/handler.go (hunk 1 above)
Comment (cnlangzi) [reply code/go/0chain.net/blobbercore/handler/handler.go:161]: yes. it was by designed. it is copied from old `WithReadonlyConnection` method

**Comment by cnlangzi** [comment]
> Do we have system test that cover this case

added it on https://github.com/0chain/system_test/pull/226

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/getobjecttree  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | fix/getobjecttree  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: code/go/0chain.net/blobbercore/handler/handler.go (hunk 1 above)
Comment (peterlimg) [reply code/go/0chain.net/blobbercore/handler/handler.go:161]: I mean if we have two API endpoints, and both call the `GetMetaDataStore().CreateTransaction()` to get the ctx, suppose endpoint A update the database, but not commit the changes yet, then another API endpoint called the rollback() in the defer function above to the same ctx, will the data changes for endpoint A be  reset?

Comment (cnlangzi) [reply code/go/0chain.net/blobbercore/handler/handler.go:161]: @peterlimg  no, each request has own context and transaction. so only changes will be rollbacked in the same request.

**Review by peterlimg — APPROVED** [review]
lgtm

**Pull request merged**